docs/source/deprecation.md

```diff
@@ -1,17 +1,516 @@
+as possible, but still it could happen. To limit such problems (and even those
+with the deprecated content), we recommend people to use APIs as much
+as possible.
+
+## What does it mean that something is deprecated?
+
+When you deprecate something, the only thing that changes is that the
+deprecated entity is marked in the code as deprecated which can have
+additional impact,likemessages produced on the user's terminal,
+in the report, ... But the rest of the functionality is the same as before,
+until the entity is removed completely.
+
+## What is the deprecation process for leapp
+
+In case some public functions, models,... (let's say, basically whatever that
+is supposed to be used by other people) are to be removed for any
+reason, these need to be marked as deprecated before the removal (if possible).
+In our projects, the deprecation will be applied only for leapp entities
+that have been introduced in an official release (IOW, a functionality that
+has been merged into the upstream, but has been removed before the release
+or was marked as experimental all the time, is going to be removed without
+the deprecation state). The time period during which the deprecated entity
+won't be removed is at least 6 months. That doesn't mean we will remove
+everything deprecated immediately after the 6 months, but it's to be expected
+that it will be dropped anytime between 6 and 12 months since the deprecation.
+
+In case of issues, deprecated entities are not going to be fixed since
+they are deprecated (unless they are fixed e.g. as a side-effect of another
+problem fix).
+
+## How does leapp informs about deprecated functionality?
+
+Mainly via release notes and changelogs. In the official leapp related projects
+(especially leapp and leapp-repository) the OAMG team take care about release
+notes to ensure they contain all data about dropped and deprecated
+functionality.
+
+Additionally, when using leapp or snactor, user is notified via messages about
+deprecated entities in **limited cases** (see below). In case of the leapp
+utility, such messages are presented inside the generated reports. In case
+of the snactor utility, the information message is printed in the console
+output in the end of the snactor execution. See examples in this page for
+detail.
+
+But be aware of limitations of our detection in leapp and snactor utilities.
+Our detection mechanisms are not bulletproof. Here is list of some examples,
+when our detection does not work well:
+
+- when an actor a message of a deprecated model, the usage of the deprecated
+  model is detected only if the message has been produced before; IOW, in
+  case the message X is produced only on machines with a specific package,
+  the deprecated usage is not detected on machines without the package
+- when a message of a deprecated model is consumed, the message points
+  internals of the leapp framework, instead of place where the message is
+  consumed, e.g.
```

Comment:
@bocekm  read the example below. Do you expect reported such a place of the usage of the deprecated entity instead. Internals of the framework instead of actor. Is it clear now? If yes, any suggestion for more clear description?